flit/wheel.py

```diff
@@ -156,12 +163,120 @@ def post_build(self):
 
         if self.upload:
             from .upload import do_upload
-            do_upload(self.wheel_file, self.metadata, self.repo)
+            do_upload(self.wheel_path, self.metadata, self.repo)
 
     def build(self):
-        self.clean_build_dir()
         self.copy_module()
         self.write_metadata()
         self.write_record()
-        self.zipup()
+
+        self.wheel_zip.close()
+
         self.post_build()
+
+
+import stat, time
+from zipfile import ZipInfo, ZIP_LZMA, _get_compressor, ZIP64_LIMIT, crc32
+
+def my_zip_write(self, filename, arcname=None, compress_type=None,
```

Comment:
I would just mark the line/section you changed in here, of add a comment with version of Python you stole that from. 
(easier for later update).

+1 otherwise. 

BTW. Flit made it to pycoders weekly [top projects of 2015](http://us4.campaign-archive2.com/?u=9735795484d2e4c204da82a29&id=22fde1c28d&e=db2b52b812)